eppo-sdk-test/store/ConfigurationStoreTest.cs

```diff
@@ -92,4 +92,53 @@ public void ShouldResetFlagsOnLoad()
 
         Assert.That(store.GetBanditFlags().Keys, Is.EquivalentTo(new List<string> { "unchangingBandit", "newBandit" }));
     }
+
+    [Test]
+    public void ShouldClearOldValuesOnSet()
+    {
+        var store = CreateConfigurationStore(new Mock<IConfigurationRequester>().Object);
+
+        var flags1 = new Flag[] {
+             new("flag1", true,new(), EppoValueType.NUMERIC, new(), 10000),
+             new("flag2", true,new(), EppoValueType.NUMERIC, new(), 10000),
+             };
+
+        var flags2 = new Flag[] {
+             new("flag1", true,new(), EppoValueType.NUMERIC, new(), 10000),
+             new("flag3", true,new(), EppoValueType.NUMERIC, new(), 10000),
+             };
+
+        store.SetConfiguration(flags1, null, null);
```

Comment:
This test doesn't cover the situation we see in the bug where the API returns a response without flags1 present. Have you considered getting away from the API of setting one flag at a time?

Reply:
> Have you considered getting away from the API of setting one flag at a time?
Do you mean [this loop?](https://github.com/Eppo-exp/dot-net-server-sdk/pull/19/files#diff-45d0635db4505504f2b1b16ec671b2ccfd525623984d81f93a884e6f1989975bR97). c#'s `MemoryCache` doesn't have an `AddMany` or similar method if that's what you mean.

Alternatively, dumping the whole dictionary into the cache under one key and doing the lookup in memory on that dict would clean up the need for multiple caches. What's the order of magnitude on how many flags a typical customer would have? If it's not in the 1000s, I think moving that lookup makes sense.

Reply:
Added a test for an empty set of flags returned. I'm not sure what the right behaviour is if the response itself is null/malformed; whether or not to flush the cache.

Reply:
> What's the order of magnitude on how many flags a typical customer would have

100s would be a HIGHLY successful account. However I see a lot of good value in the `GetConfiguration` being fast to support assignment for one particular flag.

Ok makes sense - your implementation with locking solves for the issue of having an empty cache during updating and the new unit test you wrote proves its working. thanks!